src/serialization/ParseNodeFactoryRegistry.cs

```diff
@@ -54,6 +58,43 @@ public IParseNode GetRootParseNode(string contentType, Stream content)
             if(ContentTypeAssociatedFactories.TryGetValue(cleanedContentType, out var factory))
                 return factory.GetRootParseNode(cleanedContentType, content);
 
+            throw new InvalidOperationException($"Content type {cleanedContentType} does not have a factory registered to be parsed");
+        }
+        /// <summary>
+        /// Get the <see cref="IParseNode"/> instance that is the root of the content
+        /// </summary>
+        /// <param name="contentType">The content type of the stream</param>
+        /// <param name="content">The <see cref="Stream"/> to parse</param>
+        /// <param name="cancellationToken">The cancellation token for the task</param>
+        /// <returns></returns>
+        public async Task<IParseNode> GetRootParseNodeAsync(string contentType, Stream content, 
+            CancellationToken cancellationToken = default)
+        {
+            if(string.IsNullOrEmpty(contentType))
+                throw new ArgumentNullException(nameof(contentType));
+            _ = content ?? throw new ArgumentNullException(nameof(content));
+
+            var vendorSpecificContentType = contentType.Split(";".ToCharArray(), StringSplitOptions.RemoveEmptyEntries).First();
```

Comment:
ideally the control logic should be refactored to a separate method so it can be shared between the sync and async overloads, that method would be generic and return the factory, all that would be left here would be the call to the correct get root parse node method

Reply:
Ah, ok, I see what you mean